### ISSUE TITLE
FOQUS Optimization Stop Traceback

### ISSUE THREAD

**Issue by boverhof** [CCSI-Toolset/FOQUS#603]
## Description
When running a Optimization and stopping it we get a exception.  Think this doesn't signify a logical error but should handle better for the user and should be able to detect a stop in relevant monitor thread.

## Stop
```
['Traceback (most recent call last):\n', '  File "c:\\users\\administrator\\desktop\\foqus\\foqus_lib\\framework\\optimizer\\optimization.py", line 76, in run\n    self.optimize()\n', '  File "c:\\users\\administrator\\desktop\\foqus\\foqus_lib\\framework\\optimizer\\BFGS.py", line 191, in optimize\n    options={\'ftol\':ftol, \'eps\':eps, \'maxfun\':maxeval})\n', '  File "C:\\ProgramData\\Anaconda3\\envs\\foqus\\lib\\site-packages\\scipy\\optimize\\_minimize.py", line 600, in minimize\n    callback=callback, **options)\n', '  File "C:\\ProgramData\\Anaconda3\\envs\\foqus\\lib\\site-packages\\scipy\\optimize\\lbfgsb.py", line 335, in _minimize_lbfgsb\n    f, g = func_and_grad(x)\n', '  File "C:\\ProgramData\\Anaconda3\\envs\\foqus\\lib\\site-packages\\scipy\\optimize\\lbfgsb.py", line 281, in func_and_grad\n    g = _approx_fprime_helper(x, fun, epsilon, args=args, f0=f)\n', '  File "C:\\ProgramData\\Anaconda3\\envs\\foqus\\lib\\site-packages\\scipy\\optimize\\optimize.py", line 696, in _approx_fprime_helper\n    grad[k] = (f(*((xk + d,) + args)) - f0) / d[k]\n', '  File "C:\\ProgramData\\Anaconda3\\envs\\foqus\\lib\\site-packages\\scipy\\optimize\\optimize.py", line 326, in function_wrapper\n    return function(*(wrapper_args + args))\n', '  File "c:\\users\\administrator\\desktop\\foqus\\foqus_lib\\framework\\optimizer\\BFGS.py", line 116, in f\n    objValues, cv, pv = self.prob.runSamples([x], self)\n', '  File "c:\\users\\administrator\\desktop\\foqus\\foqus_lib\\framework\\optimizer\\problem.py", line 337, in runSamples\n    return self.calculateObj(gt.res, nsamples = snum)\n', '  File "c:\\users\\administrator\\desktop\\foqus\\foqus_lib\\framework\\optimizer\\problem.py", line 393, in calculateObj\n    (x, f) = self.get_at_dicts(x,f)\n', '  File "c:\\users\\administrator\\desktop\\foqus\\foqus_lib\\framework\\optimizer\\problem.py", line 405, in get_at_dicts\n    x[i] = AtDict(x[i])\n', "TypeError: 'int' object is not iterable\n"]
```

**Comment by boverhof** [comment]
<img width="540" alt="Screen Shot 2019-10-24 at 9 40 36 AM" src="https://user-images.githubusercontent.com/738295/67506706-6ab48280-f642-11e9-94ef-57d38b886138.png">

**Comment by ksbeattie** [comment]
This is probably not an actual problem, but should be cleaned up.

**Comment by fsoepyan** [comment]
This issue was resolved with #627.
